From 76f4f3045510ff4a900c518afa22e9bf45c6841f Mon Sep 17 00:00:00 2001
From: AnotherTest <ali.mpfard@gmail.com>
Date: Sun, 18 Jul 2021 17:09:56 +0430
Subject: [PATCH 20/33] gc

---
 autohell/configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/autohell/configure.ac b/autohell/configure.ac
index 66a72a9..c476036 100644
--- a/autohell/configure.ac
+++ b/autohell/configure.ac
@@ -83,6 +83,7 @@ AS_IF([test "x$no_tls" = "xyes"], [
 ])
 AS_IF([test "x$enable_boehm_gc" = "xyes"], [
        CFLAGS+=" -DwithBoehmGC"
+       LDFLAGS+=" -lgc"
        AC_CHECK_LIB([gc], [GC_free])
 ])
 
-- 
2.32.0

